keps/sig-node/4369-allow-special-characters-environment-variable/README.md

```diff
@@ -0,0 +1,282 @@
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Allow all printable ASCII except '=', to be set as environment variables.
+
+## Motivation
+
+Kubernetes should not restrict which environment variable names can be used, because it has no way of knowing what the application may need, and people can't always choose their own variable names, which may limit the adoption of Kubernetes.
+
+### Goals
+
+* Allow users to set printable ASCII other than '=' as environment variables.
+
+### Non-Goals
+
+## Proposal
+
+* Implements relaxed validation at the top level validation method when validating API create requests.
+* Document rules for setting environment variables.
+
+#### Risks and Mitigations
+
+### User Stories (Optional)
+
+#### Story 1
+
+I am a .NET Core development engineer, .Net Core applications are using ":" when working with application settings loaded from appsettings.json file. When running .net core app in containers typically overwrite this settings by specifying environmental variable.
+such as: 
+`"Logging": { "IncludeScopes": false, "LogLevel": { "Default": "Warning" } }`    
+override like this `-e Logging:LogLevel:Default=Debug`    
+
+## Design Details
```

Comment:
The primary reason this hasn't been relaxed already is the impact on all the things currently consuming and using envvar names from the API that could break or be unsafe if the charset is expanded

https://github.com/kubernetes/kubernetes/pull/63362#issuecomment-386631005 is a good example of the types of considerations a proposal to relax validation has to consider

If envvar names are used as filenames by a third party, and assumed safe today, what happens if they contain characters that cannot be filenames (like `:`) or characters that break the assumption of a flat directory structure (like `/`)?

Reply:
Maybe we can provide enough warnings in the Api documentation to describe the risks?

For me, there seems to be no reason to use the name of envvar as the file name. I think we more often use the value of envvar as the file name, or for other purposes.

Reply:
I think these points should also be included in the Drawbacks section below.